common/phonebook.hpp

```diff
@@ -95,7 +96,7 @@ namespace ILLIXR {
 		 */
 		template <typename specific_service>
 		void register_impl(std::shared_ptr<specific_service> impl) {
-			const std::lock_guard<std::mutex> lock{_m_mutex};
+			const std::lock_guard<std::shared_mutex> lock{_m_mutex};
```

Comment:
In order to actually derive benefit from the `std::shared_mutex`, we need to change `std::lock_guard` to `std::unique_lock` in places where we _write_ to protected variables (such as this) and `std::shared_lock` in places where we _read_ (such as `lookup_impl()`).

Reply:
Changed the required locks to unique and shared for writer and reader resp.